static/js/transform-facets.js

```diff
@@ -87,6 +90,26 @@ function sortFacetOptions(options, optionsOrder, optionsFieldType, fieldId) {
       return undefined;
     }
   }
-  
-  return options.sort(applyDirectionToComparator(getSortComparator()))
+
+  return [...options].sort(applyDirectionToComparator(getSortComparator()))
 }
+
+
+/**
+ * Sorts the facet options using the priority specified in
+ * the optionsOrderList and returns a new array.
+ * 
+ * @param {{ displayName: string }[]} options The facet options to sort.
+ * @param {string[] | number[]} optionsOrderList
+ * @returns {{ displayName: string }[]}
+ */
+function sortFacetOptionsCustom(options, optionsOrderList) {
+  const getPriority = displayName => {
+    const index = optionsOrderList.indexOf(displayName);
```

Comment:
If the optionsOrderList contains numbers, do we need to convert the `displayName` first?

Reply:
I talked with Rose and she discovered that numerical fields do not support facets. Currently, if you want "numerical facets" you would use a regular string field. So, long story short we're not going to support number facets because they don't exist in the backend.